pysmt/smtlib/script.py

```diff
@@ -176,6 +177,20 @@ def serialize_to_string(self, daggify=True):
         self.serialize(buf, daggify=daggify)
         return buf.getvalue()
 
+class GoalsManager(object):
```

Comment:
Why here? I think this bit of logic should go into `shell.py` or better into a class representing a gnereic SmtLib-based OMT Solver...

Moreover, to me this class is useless, it does nothing more than a list...